### PR TITLE
add os check in discovery for aix/freenas storage and cisco cdp

### PR DESCRIPTION
**Pull request by peejaychilds**
Please give a short description what your pull request is for

When investigating another issue found errors caused by discovery trying to do things to JunOS hosts

```{"command" : ["\/usr\/bin\/snmpbulkwalk","-v2c","-c","redacted","-OQUs","-m","IBM-AIX-MIB","-M","\/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos",
"-t",2,"udp:redacted.host.net:161","aixFsTableEntry"]}
{"exitcode" : 1}
{"errorOutput" : "MIB search path: \/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos\nCannot find module (IBM-AIX-MIB): At line 1 in (none)\naixFsTableEntry: Unknown Object Identifier (Sub-id not found: (top) -> aixFsTableEntry)\n"}

{"command" : ["\/usr\/bin\/snmpbulkwalk","-v2c","-c","redacted","-OQUs","-m","FREENAS-MIB","-M","\/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos",
"-t",2,"udp:redacted.host.net:161","datasetTable"]}
{"exitcode" : 1}
{"errorOutput" : "MIB search path: \/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos\nCannot find module (FREENAS-MIB): At line 1 in (none)\ndatasetTable: Unknown Object Identifier (Sub-id not found: (top) -> datasetTable)\n"}

{"command" : ["\/usr\/bin\/snmpbulkwalk","-v2c","-c","redacted","-OQUs","-m","FREENAS-MIB","-M","\/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos",
"-t",2,"udp:redacted.host.net:161","zpoolTable"]}
{"exitcode" : 1}
{"errorOutput" : "MIB search path: \/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos\nCannot find module (FREENAS-MIB): At line 1 in (none)\nzpoolTable:
 Unknown Object Identifier (Sub-id not found: (top) -> zpoolTable)\n"}

{"command" : ["\/usr\/bin\/snmpbulkwalk","-v2c","-c","redacted","-OQUsetX","-m","CISCO-CDP-MIB","-M","\/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos","udp:redacted.host.net:161","cdpCache"]}
{"exitcode" : 1}
{"errorOutput" : "MIB search path: \/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos\nCannot find module (CISCO-CDP-MIB): At line 1 in (none)\ncdpCache: Unknown Object Identifier (Sub-id not found: (top) -> cdpCache)\n"}
```
Added device os checks

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Review by VVelox — COMMENTED** [review]
Are we sure it will always be `truenas`?

Looking at `includes/definitions/truenas.yaml`, it may also show up as `freenas` as well.

**Comment by peejaychilds** [comment]
> Are we sure it will always be `truenas`?

My read on that is that both freenas and truenas sysDescr's are considered os=truenas?

**Comment by VVelox** [comment]
> > Are we sure it will always be `truenas`?
> 
> My read on that is that both freenas and truenas sysDescr's are considered os=truenas?

Derp. Your right. That is based off the saved OS field, no the sysDescr.

File: includes/discovery/discovery-protocols.inc.php [hunk 1]
```diff
@@ -41,46 +41,49 @@
     echo PHP_EOL;
 }//end if
 
-echo ' CISCO-CDP-MIB: ';
-$cdp_array = snmpwalk_group($device, 'cdpCache', 'CISCO-CDP-MIB', 2);
+if ($device['os'] == 'ios' || $device['os'] == 'iosxe' || $device['os'] == 'nxos') {
```
Comment (PipoCanaja) [inline includes/discovery/discovery-protocols.inc.php:44]: Hi
It would make sense to check on OS group being "cisco". 
Bye

**Comment by murrant** [comment]
Hmm, other os support CDP.  Not sure about via snmp though.

**Comment by peejaychilds** [comment]
> Hmm, other os support CDP. Not sure about via snmp though.

I think most vendors that supported the Cisco proprietary have moved to LLDP - from memory only HP had the 'license' since it was proprietary.

As the MIB is in the mibs/cisco directory I assume if another device did support it that discovery would not work as the MIB would not be found (similar to what I noticed with junos devices) - so I don't think we will be breaking anything that was working before.

**Comment by PipoCanaja** [comment]
Hi @librenms/reviewers 
Tests are not impacted, but tests are also not storing the xdp neighbors as far as I know ... So we might have situations where this patch removes xdp neighbors discovery. 
What should we do with this patch ?

**Comment by peejaychilds** [comment]
> So we might have situations where this patch removes xdp neighbors discovery.

This just gates CDP discovery to Cisco devices, which is the only thing that is going to be working currently as the CDP MIB is in the cisco directory, which means even if we currently (which we do) run CDP discovery it fails due to not finding the MIB.

(as per my example which occurs during discovery on a JunOS device)
```
{"command" : ["\/usr\/bin\/snmpbulkwalk","-v2c","-c","redacted","-OQUsetX","-m","CISCO-CDP-MIB","-M","\/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos","udp:redacted.host.net:161","cdpCache"]}
{"exitcode" : 1}
{"errorOutput" : "MIB search path: \/opt\/librenms\/mibs:\/opt\/librenms\/mibs\/junos\nCannot find module (CISCO-CDP-MIB): At line 1 in (none)\ncdpCache: Unknown Object Identifier (Sub-id not found: (top) -> cdpCache)\n"}
```

**Review by murrant — APPROVED** [review]

**Pull request merged**